src/__tests__/TuxActions-test.js

```diff
@@ -87,9 +87,105 @@ describe('TuxActions', function () {
 
         tuxActionCategory.register(mockStore, actionListeners);
 
-        expect(TuxActions.register.mock.calls[0][0]).toEqual(mockStore);
+        expect(TuxActions.register.mock.calls[0][0]).toBe(mockStore);
         //expecting to find actionListeners under the CATEGORY key
-        expect(TuxActions.register.mock.calls[0][1].tests).toEqual(actionListeners);
+        expect(TuxActions.register.mock.calls[0][1].tests).toBe(actionListeners);
+      });
+
+      describe('before', function () {
+        it('should add a callback to be invoked before the corresponding action is dispatched and should receive the next callback to invoke and the passed in actionBody as inputs', function () {
+          //store a reference to the current get action and mock it out
+          var dispatchGetAction = tuxActionCategory.get = jest.genMockFunction();
+
+          var mockBeforeAction = jest.genMockFunction();
+          tuxActionCategory.before('get', mockBeforeAction);
+
+          //dispatch a get action with a mockActionBody
+          var mockActionBody = {};
+          tuxActionCategory.get(mockActionBody);
+
+          //expect the mockBeforeAction to have been called with the dispatchGetAction callback and mockActionBody
+          expect(mockBeforeAction.mock.calls[0][0]).toBe(dispatchGetAction);
+          expect(mockBeforeAction.mock.calls[0][1]).toBe(mockActionBody);
+
+          //expect the dispatchGetAction not to have been called
+          expect(dispatchGetAction).not.toBeCalled();
+        });
+
+        it('should allow adding multiple callbacks and should invoke the last added callback when the action is invoked', function () {
+          //store reference to original get method for testing later on
+          var dispatchGetAction = tuxActionCategory.get;
+
+          var mockBeforeAction = jest.genMockFunction();
+          tuxActionCategory.before('get', mockBeforeAction);
+
+          //store a reference to the current get method as this method is not mockBeforeAction but is instead the output of calling .bind on mockBeforeAction
+          var mockBoundBeforeAction = tuxActionCategory.get;
+
+          var mockNewBeforeAction = jest.genMockFunction();
+          tuxActionCategory.before('get', mockNewBeforeAction);
+
+          //dispatch a get action with a mockActionBody
+          var mockActionBody = {};
+          tuxActionCategory.get(mockActionBody);
+
+          //expect the mockNewBeforeAction to have been called with the mockBoundBeforeAction callback and mockActionBody
+          expect(mockNewBeforeAction.mock.calls[0][0]).toBe(mockBoundBeforeAction);
+          expect(mockNewBeforeAction.mock.calls[0][1]).toBe(mockActionBody);
+
+          //expect the mockBeforeAction not to be called
+          expect(mockBeforeAction).not.toBeCalled();
+
+          //invoke the first argument passed into the mockNewBeforeAction
```

Comment:
Recommend changing comments to one line reading as such: "//invoke the first argument passed into the mockNewBeforeAction and pass in a newMockActionBody as the input."